src/scripts/migrate.py

```diff
@@ -0,0 +1,201 @@
+    todo: owned_terms: if slow, can speed this up by grouping prefixes by splitting on : and filter out
+    """
+    # Get all terms: CURIES or URIs
+    terms = [x for x in ontology.entities()]
+
+    # Get CURIES
+    uri_terms = []
+    curie_terms_1 = []
+    for t in terms:
+        if t.startswith('http'):
+            uri_terms.append(t)
+        else:
+            curie_terms_1.append(t)
+
+    # - Convert from URI if needed
+    uri_converter = curies.Converter.from_prefix_map(owned_prefix_map)
+    curie_terms_2 = []
+    terms_with_no_known_prefix = []
+    for uri in uri_terms:
+        curie = uri_converter.compress(uri)
+        if curie:
+            curie_terms_2.append(curie)
+        else:
+            terms_with_no_known_prefix.append(uri)
+    curie_terms = curie_terms_1 + curie_terms_2
+
+    # Filter: Terms 'owned' by ontology
+    owned_terms = [x for x in curie_terms if any([x.startswith(y) for y in owned_prefix_map.keys()])]
+
+    return owned_terms
+
+
+def run(ontology_path: str, onto_config_path: str, sssom_map_path: str, min_id: str, mondo_terms_path: str, outpath: str) -> pd.DataFrame:
+    """Run slurp pipeline for given ontology"""
+    # Read inputs
+    # todo's: Ontology implementation selection
+    #  i. If trouble, can try `SparqlImplementation`, but ~6 min to load and queries slow (cuz rdflib)
+    #  ii. use Sql > Pronto when .entities() fixed: https://github.com/INCATools/ontology-access-kit/issues/235
+    with open(onto_config_path, 'r') as stream:
+        onto_config = yaml.safe_load(stream)
+    # ontology = SqlImplementation(OntologyResource(slug=ontology_path, local=True))
+    ontology = ProntoImplementation(OntologyResource(slug=ontology_path, local=True))
+    sssom_df = pd.read_csv(sssom_map_path, comment='#', sep='\t')
+    mondo_termlist_df = pd.read_csv(mondo_terms_path, comment='#', sep='\t')
+
+    # Get relationships
+    edge_triples_uris_and_curies: List[Tuple] = [x for x in ontology.all_relationships()]
+    # - If is URI, try to convert to CURIE
+    edge_triples_curies: List[Tuple] = []
+    owned_prefix_map: Dict[str, str] = onto_config['base_prefix_map']
+    uri_converter = curies.Converter.from_prefix_map(owned_prefix_map)
+    for triple in edge_triples_uris_and_curies:
+        new_triple = ()
+        for uri_or_curie in triple:
+            curie = uri_converter.compress(uri_or_curie)
+            new_triple += (curie,) if curie else (uri_or_curie,)
+        edge_triples_curies.append(new_triple)
+    # - Build lookup dict of all the direct parents of each term. Only includes direct parents owned by the ontology.
+    term_parent_map = {}
+    owned_prefixes = set(onto_config['base_prefix_map'].keys())
+    for triple in edge_triples_curies:
+        subject, predicate, object = triple
+        if subject not in term_parent_map:
+            term_parent_map[subject] = []
+        if predicate == 'rdfs:subClassOf' and object.split(':')[0] in owned_prefixes:
```

Comment:
@matentzn In doing my first run of this, I observed cases where parents of a term are defined outside the ontology itself. For example, in OMIM, many terms are `rdfs:subClassOf` [`SO:0000704`](http://www.sequenceontology.org/browser/current_svn/term/SO:0000704).

I think that the intention of slurp is that it only cares about terms that are owned by the given ontology, so that is how I have designed this. E.g. `SO:0000704` may be a direct parent of some OMIM terms, but it is not a slurp candidate.

I think this is correct, but if I am wrong please let me know.